lib/config/options/index.ts

```diff
@@ -722,6 +722,18 @@ const options: RenovateOptions[] = [
     subType: 'string',
     stage: 'repository',
   },
+  {
+    name: 'gitTimeout',
+    description:
+      'Configure the timeout with a number of milliseconds to wait for a git task',
+    type: 'object',
+    globalOnly: true,
+    default: {
+      timeout: {
+        block: 10000,
+      },
```

Comment:
@viceice do you think we really need to expose the whole object, or could we just let one particular field be configurable?

Reply:
Lets only create a type of number field that will hold the gitTimeout and the object for the config will be at git/config.ts?

Reply:
Yes, I think our config option should be a numeric value and not an object. The value then goes into `block`. We should also sanity check within our config validation logic that it's within a certain range, e.g. 2000-60000.

Reply:
my idea was to have a `timeouts` block where on child is `git-block`, so we can easily add more timeouts, like `http-download` ...

Reply:
I would prefer to keep it simple with separate numerical config options at the top level of global config. How many would we have. They each need documenting, default values, etc.

Reply:
Can we keep separate for now (eg call a new numerical option gitTimeout) and consolidate them later if there's too many?

Reply:
To make a separate object in the git/config.ts that will hold this timeout field value and then we will combine them where needed ?